--- origsrc/libhdate-1.4.12/bindings/php/Makefile.am	2008-06-14 02:59:23.000000000 -0500
+++ src/libhdate-1.4.12/bindings/php/Makefile.am	2008-06-15 16:14:28.890625000 -0500
@@ -1,26 +1,21 @@
-
-EXTRA_DIST = hdate.i hdate.ini
+EXTRA_DIST = hdate.i
 
 if WITH_PHP
 
-DEPS= $(top_builddir)/src/libhdate.la
+phpconfigdir = $(PH_CONFIG_PATH)
+phpconfig_DATA = hdate.ini
 
-CLEANFILES = *.o *.cpp *.h *.so *wrap*
+BUILT_SOURCES = hdate_wrap.cpp
 
-all-am: hdate.so
-	
+phpmoddir = $(PH_MODULES_PATH)
+phpmod_LTLIBRARIES = hdate.la
 
-hdate.so: hdate_wrap.cpp
-	g++ -I$(top_srcdir)/src `php-config --includes` -fpic -c hdate_wrap.cpp
-	gcc -shared ../../src/.libs/libhdate.so hdate_wrap.o -lstdc++ -o hdate.so
+hdate_la_SOURCES = hdate_wrap.cpp
+hdate_la_CPPFLAGS = -I$(top_srcdir)/src `php-config --includes`
+hdate_la_LDFLAGS = -module -avoid-version -no-undefined
+hdate_la_LIBADD = $(top_builddir)/src/libhdate.la -lphp5lib
 
 hdate_wrap.cpp: hdate.i
-	swig -php5 -c++ hdate.i
+	swig -php5 -c++ -outdir . $<
 
-install-data-local: hdate.so hdate.ini
-	$(mkinstalldirs) $(DESTDIR)$(PH_MODULES_PATH)
-	$(INSTALL_DATA) hdate.so $(DESTDIR)$(PH_MODULES_PATH)/hdate.so
-	
-	$(mkinstalldirs) $(DESTDIR)$(PH_CONFIG_PATH)
-	$(INSTALL_DATA) hdate.ini $(DESTDIR)$(PH_CONFIG_PATH)/hdate.ini
 endif
--- origsrc/libhdate-1.4.12/configure.in	2008-06-15 02:51:27.000000000 -0500
+++ src/libhdate-1.4.12/configure.in	2008-06-15 16:23:02.375000000 -0500
@@ -185,12 +188,12 @@
   [path to php site lib])])
 
 if test "$have_php" = "yes"; then
-	  AC_CHECK_PROG(have_php_config, "php-config5", yes, no)
+	  AC_CHECK_PROG(have_php_config, "php-config", yes, no)
 	  if test "$have_php_config" = "yes"; then
 	      
 	      PH_CONFIG_PATH=`php --ini | head -n1 | cut -d: -f2 | sed 's/\s//g'`
 		  if test "$with_php_sitelib_dir" = "" ; then
-		    PH_MODULES_PATH=`php-config5 --extension-dir`
+		    PH_MODULES_PATH=`php-config --extension-dir`
 		  else
 		    PH_MODULES_PATH=$with_php_sitelib_dir
 		  fi
