--- origsrc/libhdate-1.4.12/src/Makefile.am	2008-06-14 02:59:22.000000000 -0500
+++ src/libhdate-1.4.12/src/Makefile.am	2008-06-15 16:14:29.187500000 -0500
@@ -13,9 +13,9 @@
 	hdate.h\
 	support.h
 
-libhdate_la_LDFLAGS = $(libhdate_LDFLAGS) -version-info $(VERSION_INFO)
+libhdate_la_LDFLAGS = $(libhdate_LDFLAGS) -version-info $(VERSION_INFO) -no-undefined
 
-libhdate_la_LIBADD = 
+libhdate_la_LIBADD = $(LTLIBINTL)
 
 include_HEADERS = hdate.h hdatepp.h
 
