--- origsrc/libhdate-1.4.12/bindings/python/Makefile.am	2008-06-14 02:59:22.000000000 -0500
+++ src/libhdate-1.4.12/bindings/python/Makefile.am	2008-06-15 16:14:28.937500000 -0500
@@ -3,23 +3,19 @@
 
 if WITH_PYTHON
 
-DEPS= $(top_builddir)/src/libhdate.la
+pythondir = $(PY_MODULES_PATH)
+python_DATA = hdate.py
 
-CLEANFILES = *.o *.cxx *.so *wrap* *.py
+BUILT_SOURCES = hdate_wrap.cxx hdate.py
 
-all-am: _hdate.so
-	
+python_LTLIBRARIES = _hdate.la
 
-_hdate.so: hdate_wrap.cxx
-	g++ -fpic -c hdate_wrap.cxx $(PY_CFLAGS) -I$(top_srcdir)/src
-	g++ -shared ../../src/.libs/libhdate.so hdate_wrap.o -o _hdate.so
+_hdate_la_SOURCES = hdate_wrap.cxx
+_hdate_la_CXXFLAGS = -I$(top_srcdir)/src $(PY_CFLAGS)
+_hdate_la_LDFLAGS = -module -avoid-version -no-undefined
+_hdate_la_LIBADD = $(top_builddir)/src/libhdate.la $(PY_LIBS)
 
-hdate_wrap.cxx: hdate.i
-	swig -python -c++ hdate.i
-
-install-data-local: _hdate.so
-	$(mkinstalldirs) $(DESTDIR)$(PY_MODULES_PATH)
-	$(INSTALL_DATA) _hdate.so $(DESTDIR)$(PY_MODULES_PATH)/_hdate.so
-	$(INSTALL_DATA) hdate.py $(DESTDIR)$(PY_MODULES_PATH)/hdate.py
+hdate_wrap.cxx hdate.py: hdate.i
+	swig -python -c++ -outdir . $<
 
 endif
--- origsrc/libhdate-1.4.12/configure.in	2008-06-15 02:51:27.000000000 -0500
+++ src/libhdate-1.4.12/configure.in	2008-06-15 16:23:02.375000000 -0500
@@ -104,6 +105,7 @@
 	  if test -f $PY_PREFIX/include/python$PY_VERSION/Python.h; then
 		  AC_MSG_RESULT(yes)
 		  PY_CFLAGS="-I$PY_PREFIX/include/python$PY_VERSION"
+		  PY_LIBS="-L$PY_PREFIX/lib/python$PY_VERSION/config -lpython$PY_VERSION"
 		  
 		  if test "$with_python_sitelib_dir" = "" ; then
 		    PY_MODULES_PATH="$PY_PREFIX/lib/python$PY_VERSION/site-packages"
@@ -113,6 +115,7 @@
 		  
 		  AC_SUBST([PY_VERSION])
 		  AC_SUBST([PY_CFLAGS])
+		  AC_SUBST([PY_LIBS])
 		  AC_SUBST([PY_MODULES_PATH])
 		  if test "$have_swig" = "no"; then
 			build_python=no
