--- origsrc/libhdate-1.4.12/bindings/perl/Makefile.am	2008-06-14 02:59:22.000000000 -0500
+++ src/libhdate-1.4.12/bindings/perl/Makefile.am	2008-06-15 16:14:28.859375000 -0500
@@ -3,23 +3,19 @@
 
 if WITH_PERL
 
-DEPS= $(top_builddir)/src/libhdate.la
+perldir = $(PE_MODULES_PATH)
+perl_DATA = hdate.pm
 
-CLEANFILES = *.o *.cxx *.so *wrap* *.pm
+BUILT_SOURCES = hdate_wrap.cxx hdate.pm
 
-all-am: hdate.so
-	
+perl_LTLIBRARIES = hdate.la
 
-hdate.so: hdate_wrap.cxx
-	g++ -fpic -c hdate_wrap.cxx $(PE_CFLAGS) -I$(top_srcdir)/src
-	g++ -shared ../../src/.libs/libhdate.so hdate_wrap.o -o hdate.so
+hdate_la_SOURCES = hdate_wrap.cxx
+hdate_la_CXXFLAGS = -I$(top_srcdir)/src $(PE_CFLAGS)
+hdate_la_LDFLAGS = -module -avoid-version -no-undefined
+hdate_la_LIBADD = $(top_builddir)/src/libhdate.la $(PE_LIBS)
 
-hdate_wrap.cxx: hdate.i
-	swig -perl -c++ hdate.i
-
-install-data-local: hdate.so
-	$(mkinstalldirs) $(DESTDIR)$(PE_MODULES_PATH)/auto/hdate/
-	$(INSTALL_DATA) hdate.so $(DESTDIR)$(PE_MODULES_PATH)/auto/hdate/hdate.so
-	$(INSTALL_DATA) hdate.pm $(DESTDIR)$(PE_MODULES_PATH)/hdate.pm
+hdate_wrap.cxx hdate.pm: hdate.i
+	swig -perl -c++ -outdir . $<
 
 endif
--- origsrc/libhdate-1.4.12/configure.in	2008-06-15 02:51:27.000000000 -0500
+++ src/libhdate-1.4.12/configure.in	2008-06-15 16:23:02.375000000 -0500
@@ -230,6 +233,7 @@
 		  AC_MSG_RESULT(yes)
 		  PE_CFLAGS="-I$PE_PREFIX/CORE "
 		  PE_CFLAGS=$PE_CFLAGS`(perl -e 'use Config; print $Config{cppflags};')`
+		  PE_LIBS="-L$PE_PREFIX/CORE -lperl"
 		  
 		  if test "$with_perl_sitelib_dir" = "" ; then
 		    PE_MODULES_PATH=`(perl -e 'use Config; print $Config{sitelib};')`
@@ -238,6 +242,7 @@
 		  fi
 		  		  
 		  AC_SUBST([PE_CFLAGS])
+		  AC_SUBST([PE_LIBS])
 		  AC_SUBST([PE_MODULES_PATH])
 		  if test "x$have_swig" = "xno"; then
 			build_perl=no
